deploy/README.md

```diff
@@ -0,0 +1,221 @@
+ 
+
+```
+deploy/
+├── README.md
+├── api-config.json
+├── capabilities-config.json
+├── ui-config.json
+├── cromwell
+│   ├── README.md
+│   ├── app-engine
+│   ├── docker-compose
+│   └── kubernetes
+└── dsub
+    ├── README.md
+    ├── app-engine
+    ├── docker-compose
+    └── kubernetes
+```
+
+The above structure shows how the deployment instructions are organized, Cromwell and dsub have their own set of
+deployment instructions and configurations respectively. From v0.2.0, each Job Manager's release contains 2 docker images, 
+e.g. UI and API(shim), once you pick a proper version(v0.2.0+), you could deploy the 2 images either on the same server
+or different servers, depends on your preferred infrastructure.
+
+
+## Configuration files
+In order to deploy Job Manager properly, you have to provide a set of config files.
+
+### UI config
+As the `ui-config.json` shows, this file contains the information required by the Job Manager UI layer:
+
+- `apiUrl` indicates the path to the API(shim) layer. If UI and API are deployed separately, this will become the 
+url to the API.
+- `clientId` should be provided if you want to enable the 
+[Google Sign-In](https://developers.google.com/identity/sign-in/web/sign-in) feature. This require you to create a 
+valid client ID within the Google cloud project which the Job Manager is deployed to.
+- `dashboardEnabled` controls whether you want to enable the dashboard view in the UI.
+
+**Note:** This file has to be mounted as `/ui/dist/assets/environments/environment.json` in the Job Manager UI docker
+ container during the deployment.
+
+### Capabilities config (required by Cromwell) <a id="capabilities-anchor"></a>
+
+dsub is using hard-coded predefined capabilities configuration for now, so this is only configurable for Cromwell. 
+As the `capabilities-config.json` shows, this file contains the information that UI can fetch from the capabilities
+endpoint of the API, which in turn controls a few behaviors of the UI layer:
+
+- `displayFields` controls which fields would be displayed as columns in the job list view. 
+- `field` is the underlying value of the job's attribute. For Cromwell, use `labels.foo` to refer to a job label `foo`.
+- `display` is the name that will be displayed in the UI.
+- `fieldType`, `validFieldValues`, `editable` and `bulkEditable` are currently only supported by Cromwell backend. 
+Both `editable` and `bulkEditable` will be treated as false unless explicitly set to true, if the field is 
+`editable`, then `fieldType` is required.
+- `commonLabels` controls if the field will become a predefined search key in the query builder.
+- `queryExtensions` represents those ExtendedQueryFields which are queryable, depends on your requirements. Usually the 
+default value should be fine.
+- `authentication` is optional. You should set the `isRequired` to `true` if you want to enable the Google OAuth 
+Sign-In ability.
+
+For Cromwell, the default config looks like:
```

Comment:
nit: to make this README shorter you could just link these example files from the deploy directory directly (I think they're the same?) Up to you

Reply:
+1 to linking out rather than copy/pasting

Reply:
Sounds better, will do!

Reply:
Actually I realized that the code snippets in the readme file serve purposes in the documentation from the file examples, will keep it there but add more description to clarify it.